src/pudl/extract/excel_test.py

```diff
@@ -0,0 +1,99 @@
+"""Unit tests for pudl.extract.excel module."""
+import unittest
+import unittest.mock as mock
+from unittest.mock import patch
+
+import pandas as pd
+
+import pudl.extract.excel as excel
+
+
+class TestMetadata(unittest.TestCase):
+    """Tests basic operation of the excel.Metadata object."""
+
+    def setUp(self):
+        """Cosntructs test metadata instance for testing."""
+        self._metadata = excel.Metadata('test')
+
+    def test_basics(self):
+        """Test that basic API method return expected results."""
+        self.assertEqual('test', self._metadata.get_dataset_name())
+        self.assertListEqual(
+            ['books', 'boxes', 'shoes'],
+            self._metadata.get_all_pages())
+        self.assertListEqual(
+            ['author', 'pages', 'title'],
+            self._metadata.get_all_columns('books'))
+        self.assertDictEqual(
+            {'book_title': 'title', 'name': 'author', 'pages': 'pages'},
+            self._metadata.get_column_map(2010, 'books'))
+        self.assertEqual(10, self._metadata.get_skiprows(2011, 'boxes'))
+        self.assertEqual(1, self._metadata.get_sheet_name(2011, 'boxes'))
+
+
+class FakeExtractor(excel.GenericExtractor):
+    """Test friendly fake extractor returns strings instead of files."""
+
+    METADATA = excel.Metadata('test')
+    BLACKLISTED_PAGES = ['shoes']
+
+    def _load_excel_file(self, year, page):
+        return f'{page}-{year}'
+
+
+def _fake_data_frames(page_name, **kwargs):
+    """Returns panda.DataFrames.
+
+    This is suitable mock for pd.read_excel method when used together with FakeExtractor.
+    """
+    fake_data = {
+        'books-2010': pd.DataFrame.from_dict(
+            {'book_title': ['Tao Te Ching'], 'name': ['Laozi'], 'pages': [0]}),
+        'books-2011': pd.DataFrame.from_dict(
+            {'title_of_book': ['The Tao of Pooh'], 'author': ['Benjamin Hoff'], 'pages': [158]}),
+        'boxes-2010': pd.DataFrame.from_dict(
+            {'composition': ['cardboard'], 'size_inches': [10]}),
+        'boxes-2011': pd.DataFrame.from_dict(
+            {'composition': ['metal'], 'size_cm': [99]}),
+    }
+    return fake_data[page_name]
+
+
+class TestGenericExtractor(unittest.TestCase):
+    """Test operation of the excel.GenericExtractor class."""
+
+    @patch('pudl.extract.excel.pd.read_excel')
+    def test_read_excel_calls(self, mock_read_excel):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=5191218)